src/Spectre.libStatistics/Math.h

```diff
@@ -380,4 +380,30 @@ std::vector<DataType> build(size_t size, Generator generator)
     return result;
 }
 
+/// <summary>
+/// Calculates differences between adjacent elements of given container.
+/// </summary>
+/// <param name="data">The container.</param>
+/// <param name="order">Order of difference operation (how many times the difference is executed).</param>
+/// <returns>Vector of calculated differences.</returns>
+ template <class DataType>
+ std::vector<DataType> diff(gsl::span<const DataType> data, uint16_t order = 1)
+ {
+    size_t length = data.length();
+    std::vector<DataType> result = std::vector<std::remove_const<DataType>::type>(data.begin(), data.end());
+
+    if (order == 0 || length < 2 || order >= length)
```

Comment:
As I agree with first part )`order == 0`, the later are strange to me. Why don't we do usual validation with exception on failure? Or at least something in the documentation, why the behavior is such...

Reply:
Alright, I'll do some exceptions.

Reply:
I wish we could ```zjeść ciastko i mieć ciastko```, but no exceptions can be thrown without branches :worried:.

Reply:
The branching here isn't that painful as it is only a single check, then performing branchless algorithm.

Reply:
Compiler predicts branches preety well :)